frontend/src/components/widgets/Slider/Slider.tsx

```diff
@@ -60,30 +60,55 @@ class Slider extends React.PureComponent<Props, State> {
 
   public componentDidMount = (): void => {
     // Attach click event listener to slider knobs.
-    if (this.sliderRef.current) {
-      const knobSelector = '[role="slider"]'
-      const knobs = this.sliderRef.current.querySelectorAll(knobSelector)
-      knobs.forEach(knob => knob.addEventListener("click", this.handleClick))
-    }
+    this.getAllSliderRoles().forEach((knob, index) => {
+      knob.addEventListener("click", this.handleClick)
+      this.setAriaValueText(knob, index)
+    })
     this.setWidgetValueImmediately({ fromUi: false })
   }
 
+  public componentDidUpdate = (): void => {
+    this.getAllSliderRoles().forEach((knob, index) => {
+      this.setAriaValueText(knob, index)
+    })
+  }
+
   public componentWillUnmount = (): void => {
     // Remove click event listener from slider knobs.
-    if (this.sliderRef.current) {
-      const knobSelector = '[role="slider"]'
-      const knobs = this.sliderRef.current.querySelectorAll(knobSelector)
-      knobs.forEach(knob =>
-        knob.removeEventListener("click", this.handleClick)
-      )
-    }
+    this.getAllSliderRoles().forEach(knob => {
+      knob.removeEventListener("click", this.handleClick)
+    })
   }
 
   private setWidgetValueImmediately = (source: Source): void => {
     const widgetId: string = this.props.element.get("id")
     this.props.widgetMgr.setFloatArrayValue(widgetId, this.state.value, source)
   }
 
+  private getAllSliderRoles = (): Element[] => {
+    if (!this.sliderRef.current) {
+      return []
+    }
+
+    const knobSelector = '[role="slider"]'
+    const knobs = this.sliderRef.current.querySelectorAll(knobSelector)
+
+    return Array.from(knobs)
+  }
+
+  private setAriaValueText = (sliderRoleRef: Element, index: number): void => {
```

Comment:
Add a short comment about why we override the aria values?

(I wish there was a React-y way of setting these aria labels but that doesn't seem exposed in BaseUI.)

Reply:
Yea. It involves setting it in React-Range. Might be a good side thing to work on and get it updated in base web.